catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTablePartitionOperations.java

```diff
@@ -0,0 +1,329 @@
+      Transform partitionInfo = getPartitionInfo(connection);
+
+      String addPartitionSqlFormat = "ALTER TABLE `%s` ADD PARTITION `%s` VALUES %s";
+      String partitionValues;
+      Partition added;
+
+      if (partition instanceof RangePartition) {
+        Preconditions.checkArgument(
+            partitionInfo instanceof Transforms.RangeTransform,
+            "Table %s is partitioned by list, but trying to add a range partition",
+            tableName);
+
+        RangePartition rangePartition = (RangePartition) partition;
+        partitionValues = buildRangePartitionValues(rangePartition);
+
+        // The partition properties actually cannot be passed into Doris, we just return an empty
+        // map instead.
+        added =
+            Partitions.range(
+                rangePartition.name(),
+                rangePartition.upper(),
+                rangePartition.lower(),
+                Collections.emptyMap());
+      } else if (partition instanceof ListPartition) {
+        Preconditions.checkArgument(
+            partitionInfo instanceof Transforms.ListTransform,
+            "Table %s is partitioned by range, but trying to add a list partition",
+            tableName);
+
+        ListPartition listPartition = (ListPartition) partition;
+        partitionValues =
+            buildListPartitionValues(
+                listPartition, ((Transforms.ListTransform) partitionInfo).fieldNames().length);
+
+        added =
+            Partitions.list(listPartition.name(), listPartition.lists(), Collections.emptyMap());
+      } else {
+        throw new IllegalArgumentException("Unsupported partition type of Doris");
+      }
+
+      try (Statement statement = connection.createStatement()) {
+        statement.executeUpdate(
+            String.format(addPartitionSqlFormat, tableName, partition.name(), partitionValues));
+        return added;
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
+  }
+
+  @Override
+  public boolean dropPartition(String partitionName) {
+    try (Connection connection = getConnection(databaseName)) {
+      getPartitionInfo(connection);
+      String dropPartitionsSql =
+          String.format("ALTER TABLE `%s` DROP PARTITION `%s`", tableName, partitionName);
+      try (Statement statement = connection.createStatement()) {
+        statement.executeUpdate(dropPartitionsSql);
+        return true;
+      }
+    } catch (SQLException e) {
+      GravitinoRuntimeException exception = exceptionConverter.toGravitinoException(e);
+      if (exception instanceof NoSuchPartitionException) {
+        return false;
+      }
+      throw exception;
+    }
+  }
+
+  private Transform getPartitionInfo(Connection connection) throws SQLException {
+    String showCreateTableSql = String.format("SHOW CREATE TABLE `%s`", tableName);
+    // Note, the connection here will be reused and should not be closed.
```

Comment:
We have utilized the connection pool, the connection will not disconnect the real connection when we use try-with-resource mechanism.

Reply:
Go it, I'll remove the note.